include/mqt-core/fomac/FoMaC.hpp

```diff
@@ -177,16 +181,36 @@ inline auto throwIfError(int result, const std::string& msg) -> void {
 }
 
 /**
- * @brief Class representing the FoMaC library.
+ * @brief Configuration structure for session authentication parameters.
+ * @details All parameters are optional. Only set the parameters needed for
+ * your authentication method. Parameters are validated when the session is
+ * constructed.
+ */
+struct SessionConfig {
+  /// Authentication token
+  std::optional<std::string> token;
+  /// Path to file containing authentication information
+  std::optional<std::string> authFile;
+  /// URL to authentication server
+  std::optional<std::string> authUrl;
+  /// Username for authentication
+  std::optional<std::string> username;
+  /// Password for authentication
+  std::optional<std::string> password;
+  /// Project ID for session
+  std::optional<std::string> projectId;
+};
+
```

Comment:
I am pretty sure that we will need to also expose the custom config parameters here as they might be necessary for certain use cases.

One particular problem that we have at the moment: The client interface does not allow to set a BASEURL because this is a DEVICE_SESSION_PARAMETER, not a SESSION_PARAMETER.
So, at the moment, there is no (direct) way that you could be setting the baseURL for a QDMI device.
At least not directly via QDMI.
In the case that we have to deal with (not as part of this PR), I am pretty sure that we can get around this by making use of environment variables, which isn't necessarily pretty, but should work.